### ISSUE TITLE
Wizard: Mask credentials 

### ISSUE THREAD

**Issue by Haarolean** [kafbat/kafka-ui#657]
### Discussed in https://github.com/kafbat/kafka-ui/discussions/647

<div type='discussions-op-text'>

<sup>Originally posted by **AayushSaini101** November  6, 2024</sup>
<img width="1176" alt="Screenshot 2024-11-05 at 11 01 22 AM" src="https://github.com/user-attachments/assets/9b0195dd-cdd3-4bb8-939b-0fcc26cd6d14">
Hi team, we are running kafka ui in the docker and in future planning to host it so that other team can use it, but the credentials are clearly visible is there way to mask the credentials thanks </div>

**Comment by AayushSaini101** [comment]
@Haarolean Can you please prioritize this issue ?

**Comment by germanosin** [comment]
@autometa101 Thanks for a great point. Disabling dynamic configuration will prevent users from viewing secrets and credentials, but it will require using static configurations through server environment variables or a config file.

Masking secret fields is not a straightforward task, as secrets can appear in unstructured strings or properties, which is quite common in Kafka configurations.

In the future, I’d like to enable runtime secret retrieval from secure sources like Vault or AWS Secrets Manager. However, this would require significant effort, so I plan to postpone this feature and carefully consider its implementation before moving forward.

**Comment by autometa101** [comment]
@germanosin Yes, I use Terraform to pass configuration and avoid setting dynamic properties.
P.S. Masking secret fields is absolutely essential for production servers.
